### PR TITLE
Add resource group to some object formats

### PR DESCRIPTION
**Pull request by rynowak**
# Description

This change refactors our code a bit for object formats used in the tables displayed by the CLI. This also adds resource group to some of the more commonly used formats. I did some general consistency cleanup and added tests where they were missing.

This is part of an effort to make resource groups more usable as a concept.

## Type of change

- This pull request adds or changes features of Radius and has an approved issue (issue link required).

Part of: #6827

### Summary
📝🧪♻️

This pull request refactors and updates the CLI output formatting logic for various commands, such as status, list, show, and update. It introduces new functions and types in separate files under `pkg/cli/cmd` to handle the formatting for different objects, such as applications, gateways, credentials, environments, and groups. It also adds license header comments and test package declarations to the new files, and updates the unit tests and imports accordingly.

> _We're the masters of the CLI_
> _We format the output with our skill_
> _We refactor and update our code_
> _We make the license header bold_

### Walkthrough
*  Add license header comments and package declarations to new files `objectformats.go` and `objectformats_test.go` in various subpackages of `pkg/cli/cmd` ([link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-e06a44cea8361e2d18e8364dc3bd4314addb8d06363a34c55c1580c553b87b8aR1-R54), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-68a2f8c98489dcc7a323cf0d207e0e1bbd1c0e2ed92ff78ab349f63b23097a02R1-R42), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-52a83605230057a2497b7fe3c6a72535ed246483766377aa4b9a47dea8da455fR1-R66), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-5dc5d1caf78012566c3dd84a46d6822d122f6a355b8d4a77a05035f41a48a786R1-R51), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-2ac3287b3b738308dcfb5d03e0719b4ecf949bd4ce29ef81f42fba20b36919baR1-R41), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-0c16bfbe9fa55b51aa34d6f26bd4ed0b0eb2677e8a4c75c15f0e85209acd6bbbR1-R35), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-c4f775f7262efd1dcf8eee0ed5b70793cd0e428a17909488a415ff51569bc887R1-R41))
* Move and rename functions that format the output of various commands from `objectformats.go` to the respective files where they are used, and update the corresponding calls and tests ([link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d3a3459434e925881f5cd87077b6f7db62ff30d190094cb46a71d632a71fe32dL189-R188), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d3a3459434e925881f5cd87077b6f7db62ff30d190094cb46a71d632a71fe32dL197-R196), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-2b3a46be3fab0d68fc258c962cf6ce1621b93ca7667fb4239a2ba6e0d1ec3308L173-R172), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-2b3a46be3fab0d68fc258c962cf6ce1621b93ca7667fb4239a2ba6e0d1ec3308L182-R181), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-926e301add91949c5a44382390c0b2c18616911beb4796e72b57aac0cc2ff2dcL115-R114), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-22afbf3787c0875661e06987c8d9d419cc8e91384f199ec8f44cca77afcdd146L113-R112), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d7e98275cadf1da7d21c9ac06cdda7907310f6152077c0db4096ab3264446460L144-R143), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d7e98275cadf1da7d21c9ac06cdda7907310f6152077c0db4096ab3264446460L211-R210), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d4a9101ceaa220d437b0ef29440cbba8d20178502d94aa88eb33524bbe48bdfbL129-R128), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-c8b9a3315a934ca0d8e0b750b2872603b1c504239f5d310251e87e5fb6a8e4a2L353-R352), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-c8b9a3315a934ca0d8e0b750b2872603b1c504239f5d310251e87e5fb6a8e4a2L529-R528), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-5666f913c4913108ef57c2f23dec3f4b6a4fc6bc653f2b0c51982b9619fb6e12L263-R262), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-b6f63f3ed2aaeada863bd788b630879282e34fce2ad6672636ea677a37748853L134-R134), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-260f1d886aae571f8487e5405785053daf07362bfef8b8bc6d9509900a3b3cc0L122-R122))
* Remove unused imports of the package `objectformats` from various files in `pkg/cli/cmd` ([link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d3a3459434e925881f5cd87077b6f7db62ff30d190094cb46a71d632a71fe32dL30), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-2b3a46be3fab0d68fc258c962cf6ce1621b93ca7667fb4239a2ba6e0d1ec3308L28), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-926e301add91949c5a44382390c0b2c18616911beb4796e72b57aac0cc2ff2dcL27), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-22afbf3787c0875661e06987c8d9d419cc8e91384f199ec8f44cca77afcdd146L27), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d7e98275cadf1da7d21c9ac06cdda7907310f6152077c0db4096ab3264446460L28), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-d4a9101ceaa220d437b0ef29440cbba8d20178502d94aa88eb33524bbe48bdfbL28), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-c8b9a3315a934ca0d8e0b750b2872603b1c504239f5d310251e87e5fb6a8e4a2L32), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-5666f913c4913108ef57c2f23dec3f4b6a4fc6bc653f2b0c51982b9619fb6e12L30), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-b6f63f3ed2aaeada863bd788b630879282e34fce2ad6672636ea677a37748853L25-R27), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-260f1d886aae571f8487e5405785053daf07362bfef8b8bc6d9509900a3b3cc0L24-R26))
* Add import of the package `common` to the files `pkg/cli/cmd/group/list/list.go` and `pkg/cli/cmd/group/list/list_test.go`, which use the function `common.ResourceGroupFormat()` ([link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-b6f63f3ed2aaeada863bd788b630879282e34fce2ad6672636ea677a37748853L25-R27), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-260f1d886aae571f8487e5405785053daf07362bfef8b8bc6d9509900a3b3cc0L24-R26))
* Replace the type `objectformats.OutputEnvObject` with `environmentForDisplay` in the files `pkg/cli/cmd/env/update/update.go` and `pkg/cli/cmd/env/update/update_test.go`, as the former type has been moved to the latter file and renamed ([link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-c8b9a3315a934ca0d8e0b750b2872603b1c504239f5d310251e87e5fb6a8e4a2L339-R339), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-c8b9a3315a934ca0d8e0b750b2872603b1c504239f5d310251e87e5fb6a8e4a2L516-R516), [link](https://github.com/radius-project/radius/pull/6894/files?diff=unified&w=0#diff-5666f913c4913108ef57c2f23dec3f4b6a4fc6bc653f2b0c51982b9619fb6e12L256-R256))

### CODE REVIEW

File: pkg/cli/cmd/app/status/objectformats.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import "github.com/radius-project/radius/pkg/cli/output"
+
+// statusFormat sets up the columns and headings for a table to display application names and resource counts.
+func statusFormat() output.FormatterOptions {
```
Comment (rynowak) [inline pkg/cli/cmd/app/status/objectformats.go:22]: I moved formats to the most specific package that I could. A lot of these formats were defined to output tables for private data types, which meant they could not be tested without **some** kind of refactor. 

It made way more sense to keep the functionality in the packages that need it.

File: pkg/cli/cmd/recipe/types.go [hunk 2]
```diff
@@ -24,3 +24,11 @@ type EnvironmentRecipe struct {
...
 	PlainHTTP       bool   `json:"plainHTTP"`
 }
+
+type RecipeParameter struct {
+	Name         string      `json:"name,omitempty"`
+	DefaultValue interface{} `json:"defaultValue,omitempty"`
+	Type         string      `json:"type,omitempty"`
+	MaxValue     string      `json:"maxValue,omitempty"`
+	MinValue     string      `json:"minValue,omitempty"`
+}
```
Comment (rynowak) [inline pkg/cli/cmd/recipe/types.go:34]: Needed to bring this up a level so I could use it in these tests.

**Comment by rynowak** [comment]
**After:**

image

**Review by ytimocin — APPROVED** [review]

**Pull request merged**